### PR TITLE
[7.2.x] Scala 2.13.0-M3

### PR DESCRIPTION
**Pull request by xuwei-k**
<del>partially(?)</del>  fix https://github.com/scalaz/scalaz/issues/1629
(can't build some modules due to https://github.com/rickynils/scalacheck/issues/382 )

### CODE REVIEW

File: core/src/main/scala/scalaz/Adjunction.scala [hunk 1]
```diff
@@ -117,7 +117,7 @@ sealed abstract class AdjunctionInstances {
     }
 
   implicit def writerReaderAdjunction[E]: Adjunction[Writer[E, ?], Reader[E, ?]] =
-    new Adjunction[Writer[E, ?], Reader[E, ?]] {
+    new Adjunction[Writer[E, ?], Reader[E, ?]]()(WriterT.writerTraverse[E], implicitly) {
```
Comment (joroKr21) [inline core/src/main/scala/scalaz/Adjunction.scala]: The real solution is to reorder the implicits in `WriterT`. I wonder why this didn't come up in #1632 (maybe something changed between 7.2.x and 7.3.x). I can help with this after scalacheck is out.

File: core/src/main/scala/scalaz/Free.scala [hunk 2]
```diff
@@ -357,10 +357,14 @@ object Trampoline extends TrampolineInstances {
 
   def suspend[A](a: => Trampoline[A]): Trampoline[A] =
     Free.suspend(a)
+
+  implicit def trampolineComonad: Comonad[Trampoline] =
+    trampolineInstance
```
Comment (joroKr21) [inline core/src/main/scala/scalaz/Free.scala]: Neat, but why is it defined twice?

Comment (xuwei-k) [reply core/src/main/scala/scalaz/Free.scala]: removed

File: core/src/main/scala/scalaz/Adjunction.scala (hunk 1 above)
Comment (xuwei-k) [reply core/src/main/scala/scalaz/Adjunction.scala]: fixed

File: tests/src/test/scala/scalaz/FreeTTest.scala [hunk 3]
```diff
@@ -115,7 +115,7 @@ object FreeTTest extends SpecLite {
 
   private def compilationTest = {
     val a: String \/ Int = \/-(42)
-    val b: FreeT[Maybe, String \/ ?, Int] = FreeT.liftMU(a)
+    val b: FreeT[Maybe, String \/ ?, Int] = FreeT.liftMU[Maybe, String \/ Int](a)
```
Comment (xuwei-k) [inline tests/src/test/scala/scalaz/FreeTTest.scala]: :confused:

**Pull request merged**